plugins/catalog-backend-module-aws-accounts/src/providers/AwsOrganizationProvider.ts

```diff
@@ -0,0 +1,255 @@
+        if (!providerConfig.schedule) {
+          throw new Error('Schedule is required');
+        }
+        const taskRunner = options.scheduler.createScheduledTaskRunner(
+          providerConfig.schedule,
+        );
+
+        const awsCredentialsManager: AwsCredentialsManager =
+          options.awsCredentialsManager ||
+          DefaultAwsCredentialsManager.fromConfig(config);
+
+        const credProvider = await awsCredentialsManager.getCredentialProvider({
+          accountId: providerConfig.accountId,
+        });
+
+        return new AwsOrganizationEntityProvider(
+          providerConfig,
+          options.logger,
+          taskRunner,
+          credProvider,
+        );
+      }),
+    );
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    return await this.scheduleFn();
+  }
+
+  async refresh(logger: LoggerService) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    logger.info('Discovering AWS Organization AWS accounts');
+
+    let accounts: Account[];
+
+    try {
+      accounts = await this.getAwsAccounts();
+    } catch (e: any) {
+      this.logger.error('error fetching AWS Organization accounts', e);
+      return;
+    }
+
+    const resources = accounts.map(c => this.accountToResource(c));
+
+    logger.info(`Discovered ${resources.length} AWS Organization accounts`);
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: resources,
+    });
+
+    logger.info('Finished syncing AWS Organization accounts');
+  }
+
+  private async getAwsAccounts(): Promise<Account[]> {
+    const awsAccounts: Account[] = [];
+
+    for await (const page of paginateListAccounts(
+      { client: this.organizations },
+      {},
+    )) {
+      awsAccounts.push(...(page.Accounts || []));
+    }
+
+    return awsAccounts;
+  }
+
+  private normalizeName(name: string): string {
+    return name
+      .trim()
+      .toLocaleLowerCase('en-US')
+      .replace(/[^a-zA-Z0-9\-]/g, '-');
+  }
+
+  private normalizeAccountStatus(name: string): string {
+    return name.toLocaleLowerCase('en-US');
+  }
+
+  private extractInformationFromArn(arn: string): {
+    accountId: string;
+    organizationId: string;
+  } {
+    const parts = arn.split('/');
+
+    return {
+      accountId: parts[parts.length - 1],
+      organizationId: parts[parts.length - 2],
+    };
+  }
+
+  private accountToResource(account: Account): DeferredEntity {
+    if (!account.Arn) {
+      throw new Error('Account ARN is missing');
+    }
+
+    const { accountId, organizationId } = this.extractInformationFromArn(
+      account.Arn,
+    );
+
+    const entity: ResourceEntityV1alpha1 = {
+      apiVersion: 'backstage.io/v1alpha1',
+      kind: 'Resource',
+      metadata: {
+        annotations: {
+          [ACCOUNTID_ANNOTATION]: accountId,
+          [ARN_ANNOTATION]: account.Arn || '',
```

Comment:
You may want to actually take care to avoid all these empty strings; a lot of places in the model explicitly reject empty strings as illegal values. If you set them to undefined, you can wrap objects in lodash's `pickBy` to filter out only values that actually have some content https://github.com/backstage/backstage/blob/732583700dd0809dae4614792bd67177d7c9246e/plugins/auth-backend/src/identity/KeyStores.ts#L60